src/q7/algodata-LINMA2472/summary/algodata-LINMA2472-summary.tex

```diff
@@ -0,0 +1,1033 @@
+
+\subsubsection{Erdos-Renyi}
+
+\begin{itemize}
+    \item Fix number of nodes n
+    \item Fix number of (undirected) edges m
+    \item We have an uniform distribution on all graphs G(n,m)
+\end{itemize}
+
+We might be interested by a variant (\textbf{Gilbert}) where we fix p as the probability of being friend with someone. This is interesting because we can analyze the degree distribution :
+
+\begin{quote}
+Basically, if we have a probability of becoming friends with another, then we are in the case of a binomial distribution of mean $p(n-1) \approx pn $. So, a good approximation of that is a poisson where $\lambda = pn$ and $p \ll 1$. And by the CLT the poisson itself can be approximated by a gaussian if $pn \gg 1$.
+\end{quote}
+
+We discover some new properties: 
+\begin{itemize}
+    \item \textbf{Monotone increase property} : if you add edges, the properties are still valid.
+    \item \textbf{Every monotone property has a threshold function $p^*(n)$} : \\
+    If $\frac{p(n)}{p^*(n)} = 0$ then Prob(x) $\to 0$ and if conversely $p(n)$ is far above $p^*(n)$ we have the Prob(x) $\to 1$. \\
+    As an exemple : for ``at least one edge", we have the threshold function $p^*(n) = 1/n^2$. \\
+    $$\mathbb{E}(\sharp\{\text{edges}\}) = \Big( \begin{array}{c} n \\ 2 \end{array}\Big) p \sim n^2p \Rightarrow \begin{array}{c} 
+    p \ll p^* \quad \text{then} \quad \mathbb{E}(\sharp\{\text{edges}\}) \ll 1 \\
+    p \gg p^* \quad \text{then} \quad \mathbb{E}(\sharp\{\text{edges}\}) \gg 1
+    \end{array}
+    $$
+\end{itemize}
+
+Note that we do not have heterogeneous degrees or density or many triangles : $\mathbb{E}(\sharp\{\bigtriangledown\}) = \Big( \begin{array}{c} n \\ 3 \end{array} \Big) p^3 = \Theta\big((pn)^3\big) = \Theta\big(\bar{d^3}\big)$
+
+The last thing to remember about ER random graph is that the diameter and average distance are close to $ln(n)$ and $ln(pn) \sim ln(d)$. Basically this act kind of an `epidemic' with $log_d n$ generations to hit the whole component.
+\subsubsection{Configuration}
+
+In order to improve the previous models and make them more realistic, several approaches are discussed. 
+
+\begin{itemize}
+    \item \textbf{Fixing degree of each node}: one consider that the degree distribution is very wrong and impose a degree. 
+    \item \textbf{Stub model}: we randomly wire defined stubs.
+    \item \textbf{Random swaps}: taking an existing network and randomly swap edges.
+\end{itemize}
+
+This configuration model is similar in many aspects to ER random graphs, for example it still have an homogeneous density and a low diameter log n / log D where $D = \frac{\sum d_i^2}{\sum d_i}$ (but not homogeneous degree anymore).
+
+
+\subsubsection{Watts-Strogatz (small world)}
+
+The basic idea behind this model is that most friends live nearby but it happens that some live far-away. We start with a circle and then randomly rewire each link with a probability p. A variant is adding links instead of editing them. \\
```

Comment:
```suggestion
The basic idea behind this model is that most friends live nearby but it happens that some live far-away. We start with a circle and then randomly rewire each link with a probability p. A variant is adding links instead of editing them.
```